File: problem/Least_Length_K/kLengthApart.py
class Solution(object):
    def kLengthApart(self, nums, k):
        
        distance = 0
        found = False
        
        for num in nums:
            if num == 1:
                
                if distance <= k and found:
                    return False 
                
                found = True

                distance = 0
            
            distance += 1

        return True

File: problem/Least_Length_K/kLengthApart.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
function kLengthApart(nums, k) {
    let lastOneIndex = -1

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === 1) {
            if (lastOneIndex !== -1 && i - lastOneIndex - 1 < k) return false
            lastOneIndex = i
        }
    }

    return true
};